### ISSUE TITLE
📣 Important note about Final Peer Evaluation (Part 1 & 2) ❗️

### ISSUE THREAD

**Issue by damithc** [nus-cs2103-AY1718S1/forum#230]
Reasons why you need to pay special attention to the final peer evaluation:
1. Unlike previous ones, final peer evaluations are **graded based on quality and accuracy**. Low quality submissions will be penalized. Besides, they can affect others' grades so try extra hard to be fair and accurate.
2. They serve as **practice for the Practical Exam** this Friday. Take your time to study and follow the evaluation criteria you need to use in the peer evaluation because you'll be using the same criteria in the Practical Exam.

If you submitted your peer evaluation earlier, please take another look at your submission due to these reasons:
1. I changed the effort estimation scale from 0-25 to 0-30, which caused the previous answers to that question to be deleted. In that case please answer that question again.
2. There was  a typo in the evaluation criteria for that question which could have affected your answer. In case you were previously mislead by the typo (my apologies), the evaluation criteria was supposed to say that **we expect most effort estimates to be below 20**. 

❗️ Note that if you estimate someone's effort to be over 20, it has to be a really stellar effort (e.g. in the range of 2-5 KLoC of code). An effort in the range of 15-20 is enough for an A grade. The reason we allow up to 30 is mainly to catch over-estimators 😈 )

On an unrelated note, project demos went well today. However, sitting through 70 demos over three days takes its toll; so, **don't feel bad if I don't appear very excited about your demo** -- it is because I'm fatigued (and a bit unwell too 🤒 ).  

On another unrelated note, I'll **send out the practice essay paper** after the project demos. I couldn't finish  setting it before the demos started.

**Comment by Kennard123661** [comment]
Hi Prof @damithc , by 2-5kLoC, is it in the main collate or the test collate or outside of these files? 

And the reference we had to estimate was based on the undo/redo mechanism, so giving a teammate 30/30 is not impossible, using the undo/redo feature as a benchmark.

**Comment by damithc** [comment]
> by 2-5kLoC, is it in the main collate or the test collate or outside of these files?

All, including documentation.

> And the reference we had to estimate was based on the undo/redo mechanism, so giving a teammate 30/30 is not impossible, using the undo/redo feature as a benchmark.

Indeed, some small percentage of students are likely to be near 30. I don't know how many because I haven't looked at the code myself (I'm going by past experience only). So, if you are giving higher than 20, you'd better be sure because it will be scrutinized extra carefully by us.

BTW, undo/redo is a rather complex feature (as you can see from the various alternatives considered), affecting all other current/future commands, requiring a redesign of the command inheritance structure, and it took a lot of effort to make it not look so complex.

**Comment by hansiang93** [comment]
Hi prof @damithc, so if we're getting this right, the team would be looking at total effort rather than just the code contributed in collate? So things like deployment, QA checking, ensuring meeting of deadlines etc would be included?

Could i ask how would the team be evaluating these non-code efforts?

**Comment by damithc** [comment]
> Hi prof @damithc, so if we're getting this right, the team would be looking at total effort rather than just the code contributed in collate? So things like deployment, QA checking, ensuring meeting of deadlines etc would be included?
> 
> Could i ask how would the team be evaluating these non-code efforts?

Yes, we'll have to take all those into account as far as possible. And we have to base our estimate on information provided in the project portfolio page.
